table/format.cc

```diff
@@ -357,49 +369,64 @@ Status ReadBlockContents(RandomAccessFileReader* file, const Footer& footer,
     }
   }
 
-  if (cache_options.persistent_cache &&
-      cache_options.persistent_cache->IsCompressed()) {
+  bool got_from_prefetch_buffer = false;
```

Comment:
If your concern was to check the prefetch_buffer before checking the persistent cache, an alternative API would have been to add pass a no_io option to ::Read, if it failed ty the persistent cache, and if it failed try ::Read with no_io = false. That would have the advantage of respecting the boundary between File and TableReader.